src/main/java/seedu/address/logic/UndoRedoStack.java

```diff
@@ -0,0 +1,96 @@
+package seedu.address.logic;
+
+import java.util.LinkedList;
+
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.RedoCommand;
+import seedu.address.logic.commands.UndoCommand;
+import seedu.address.logic.commands.UndoableCommand;
+
+/**
+ * Undo-redo Stack
+ * Solution adapted from https://github.com/se-edu/addressbook-level4/pull/610/files with modifications / bug fixes
+ */
+public class UndoRedoStack {
+
+    private LinkedList<UndoableCommand> undoStack;
+    private LinkedList<UndoableCommand> redoStack;
```

Comment:
Is there a reason why you opted to use LinkedList instead of Stack?

Reply:
I echo this sentiment. Based on what you're naming your variables, it should be a `Stack` instead, even though the underlying implementation of a `Stack` is based on `LinkedList`.

Reply:
Made a new commit...

Reply:
Yes, i pushed whatever that can work first...